src/services/nifSemanticValidationService.js

```diff
@@ -0,0 +1,100 @@
+services.factory('nifSemanticValidationService', [nifSemanticValidationService]); 
+
+function nifSemanticValidationService() {
+	
+	function isDNISemanticallyCorrect(dni) {
+		return isDNIControlDigitCorrect(dni);
+	}
+
+	function isNIESemanticallyCorrect(nie) {
+		var firstNIELetter = nie.charAt(0).toUpperCase();
+		var firstNIENumber;
+
+		if (firstNIELetter == 'X') {
+			firstNIENumber = '0';
+		} else if (firstNIELetter == 'Y') {
+			firstNIENumber = '1';
+		} else if (firstNIELetter == 'Z') {
+			firstNIENumber = '2';
+		}
+
+		var otherNIEchars = nie.substr(1, nie.length - 1);
+		return isDNIControlDigitCorrect(firstNIENumber + otherNIEchars);
+	}
+
+	function isDNIControlDigitCorrect(dni) {
+		var possibleControlDigits = 'TRWAGMYFPDXBNJZSQVHLCKE';
+
+        var dniNumber = dni.substr(0, dni.length - 1);
+        var dniControlDigit = dni.substr(dni.length - 1, 1).toUpperCase();
+
+        if (possibleControlDigits.charAt(dniNumber % 23) == dniControlDigit) {
+            return true;
+    	} else {
+        	return false;
+        }
+	}
+
+	function isNIFKLMSemanticallyCorrect(nifKLM) {
+		return isLegalNIFControlDigitCorrect(nifKLM);
+	}
+
+	function isLegalNIFSemanticallyCorrect(legalNIF) {
+		return isLegalNIFControlDigitCorrect(legalNIF);
+	}
+
+	function isLegalNIFControlDigitCorrect(nif) {
+		var sumA = 0;
```

Comment:
I think we can enhance readiblity on this method, please read about the algorithm and use better named variables